container/shipit/base_engine.py

```diff
@@ -2,29 +2,166 @@
 from __future__ import absolute_import
 
 import logging
-
 logger = logging.getLogger(__name__)
 
+import os
+import yaml
+import re
+import glob
+
+from .constants import SHIPIT_PATH, SHIPIT_PLAYBOOK_PREFIX, SHIPIT_ROLES_DIR
+from collections import OrderedDict
+from .utils import create_path, represent_odict
+
 
 class BaseShipItEngine(object):
+    name = None
 
-    def __init__(self, base_path):
+    def __init__(self, base_path=None, project_name=None, config=None):
         self.base_path = base_path
+        self.project_name = project_name
+        self.config = config
+        self.role_name = "%s_%s" % (self.project_name, self.name)
+        self.roles_path = os.path.join(self.base_path, SHIPIT_PATH, SHIPIT_ROLES_DIR, self.role_name)
+
+    def add_options(self, subparser):
+        """
+        Given an argument subparser, add to it the arguments and options
+        this engine allows.
+
+        https://docs.python.org/2/library/argparse.html#sub-commands
+
+        :param subparser: An argparse.ArgumentSubparser
+        :return: None
+        """
+        subparser.add_argument('--save-config', action='store_true',
+                               help=(u'Generate and save the %s configuration files in '
+                                     u'ansible/shipit_config/kubernetes.' % self.name),
+                               dest='save_config', default=False)
+
+        subparser.add_argument('--pull-from', action='store',
```

Comment:
Similar comment to the one on `push` about `pull-from` allowing either a URL or a registry entry name.